src/renderer/features/playlists/components/playlist-detail-song-list-header-filters.tsx

```diff
@@ -328,16 +338,26 @@ export const PlaylistDetailSongListHeaderFilters = ({
                                         id: playlistId,
                                         limit,
                                         startIndex,
-                                        ...filters,
+                                        ...restFilters,
                                     },
                                 }),
                             { cacheTime: 1000 * 60 * 1 },
                         );
-
-                        params.successCallback(
-                            songsRes?.items || [],
-                            songsRes?.totalRecordCount || 0,
-                        );
+                        const lowerCaseSearchTerm = searchTerm?.toLowerCase() || '';
+                        const filteredItems =
+                            songsRes?.items.filter((record) =>
+                                isAnyFieldsContainsString(
+                                    record,
+                                    SEARCH_FIELDS,
+                                    lowerCaseSearchTerm,
+                                ),
+                            ) || [];
+                        // HY: not optimal if the data is paginated. if anyone knows how to do server side filtering, then please let me know
```

Comment:
This might actually be a blocker. The workaround would be to remove server-side pagination and fetch the entire playlist at once so that client-side pagination works more effectively. I was already thinking about reworking the API to handle that, but I hadn't gotten around to it yet. I may put this PR on hold until that is implemented.